avoidance/src/transform_buffer.cpp

```diff
@@ -0,0 +1,128 @@
+#include "avoidance/transform_buffer.h"
+
+namespace avoidance {
+
+TransformBuffer::TransformBuffer(float buffer_size_s) : buffer_size_(ros::Duration(buffer_size_s)) {
+  mutex_.reset(new std::mutex);
+};
+
+std::string TransformBuffer::get_key(const std::string& source_frame, const std::string& target_frame) const {
+  return source_frame + "_to_" + target_frame;
+}
+
+bool TransformBuffer::interpolateTransform(const tf::StampedTransform& tf_earlier, const tf::StampedTransform& tf_later,
+                                           tf::StampedTransform& transform) const {
+  // check if the requested timestamp lies between the two given transforms
+  if (transform.stamp_ > tf_later.stamp_ || transform.stamp_ < tf_earlier.stamp_) {
+    return false;
+  }
+
+  ros::Duration timeBetween = tf_later.stamp_ - tf_earlier.stamp_;
+  ros::Duration timeAfterEarlier = transform.stamp_ - tf_earlier.stamp_;
+  float tau = static_cast<float>(timeAfterEarlier.toNSec()) / timeBetween.toNSec();
```

Comment:
toNSec() returns an integer. So I had to cast one of them to avoid integer division. Casting on the outside would not be good but I could cast the second one individually as well. Would that be better?

Reply:
I don't think we really care about getting rid of all the implicit casts in our codebase. To me this is fine.

Reply:
Fair enough. It just looked like a bracket error from afar, which is why I noticed. I don't care about implicit casts until they start producing compiler warnings